partiql-parser/src/main/antlr/PartiQL.g4

```diff
@@ -136,6 +137,19 @@ uniqueConstraintDef
 // but we at least can eliminate SFW query here.
 searchCondition : exprOr;
 
+// SQL Extension, Support additional table metadatas such as partition by, tblProperties, etc.
+tableExtension
+    : PARTITION BY partitionExpr                                                           # PartitionBy
+    | TBLPROPERTIES PAREN_LEFT keyValuePair (COMMA keyValuePair)* PAREN_RIGHT              # TblProperties
+    ;
+
+keyValuePair : key=LITERAL_STRING EQ value=literal;
```

Comment:
Not sure if the full `literal` is required. Maybe it can get limited to  `LITERAL_STRING` for now?

Reply:
I do not think we should permit Literal, i.e., key to be a string or timestamp does not make much sense to me. 

I think (and forget to leave a comment here), the question is more towards `identifier` vs `string`? 

i.e., 

```
TBLPROPERTIES(myKey = 'some value')
```

vs 
```
TBLPROPERTIES('myKey' = 'someValue' )
```

Reason why this PR leans towards using string is 
1. this syntax are based from HIVE and HIVE uses string. 
2. Supporting string as key value seems reasonable to me and I don't think we need to deprecate this later. 
3. Supporting identifier as string seems to be a ergonomic feature and most likely we will process the identifier as string in subsequent process (as opposite to attempting to bind the identifier to a variable...). If you agree this is the case then we can always add the support later.

Reply:
I also agree with the key being `string`, but what I meant was do we need `value=literal`:

```
literal
    : NULL                                                                                # LiteralNull
    | MISSING                                                                             # LiteralMissing
    | TRUE                                                                                # LiteralTrue
    | FALSE                                                                               # LiteralFalse
    | LITERAL_STRING                                                                      # LiteralString
    | LITERAL_INTEGER                                                                     # LiteralInteger
    | LITERAL_DECIMAL                                                                     # LiteralDecimal
    | ION_CLOSURE                                                                         # LiteralIon
    | DATE LITERAL_STRING                                                                 # LiteralDate
    | TIME ( PAREN_LEFT LITERAL_INTEGER PAREN_RIGHT )? (WITH TIME ZONE)? LITERAL_STRING   # LiteralTime
    | TIMESTAMP ( PAREN_LEFT LITERAL_INTEGER PAREN_RIGHT )? (WITH TIME ZONE)? LITERAL_STRING   # LiteralTimestamp
    ;

```

Reply:
Oh I see. 

I am not against limiting the scope to string value only, but I don't see any drawback to support all other values. It could be useful for customer who wants to specify like `TBLPROPERTIES('isActive' = false)`.

Reply:
Perhaps would be good for us to avoid widening the scope for now :)